mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/query/parsing/impl/MongoDBBasedQueryParserService.java

```diff
@@ -32,9 +31,16 @@
 
 	private volatile SessionFactoryEntityNamesResolver entityNamesResolver;
 
+	private ThreadLocal<QueryParser> parser = new ThreadLocal<QueryParser>() {
```

Comment:
@DavideD  I changed "new QueryParser()" to "parser.get()". I think that "new QueryParser()" operation is expensive. I offer to optimize it. Are You agree?